src/EnergyPlus/General.cc

```diff
@@ -333,10 +513,23 @@ namespace General {
 			DY = Y0 - Y1;
 			if ( std::abs( DY ) < SMALL ) DY = SMALL;
 			// new estimation
-			XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			if ( ( HVACSystemRootFinding.TypeNum == Bisection ) ||
+				( HVACSystemRootFinding.TypeNum == Alternation && AltIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == RegulaFalsiThenBisection && NIte > HVACSystemRootFinding.NumOfIter ) ||
+				( HVACSystemRootFinding.TypeNum == BisectionThenRegulaFalsi && NIte <= HVACSystemRootFinding.NumOfIter ) ) {
+				if ( HVACSystemRootFinding.TypeNum == Alternation ) {
+					if ( AltIte >= 2 * HVACSystemRootFinding.NumOfIter ) AltIte = 0;
+				}
+				// Bisection
+				XTemp = ( X1 + X0 ) / 2.0;
+			} else {
+				// Regula Falsi
+				XTemp = ( Y0 * X1 - Y1 * X0 ) / DY;
+			}
```

Comment:
@lgu1234 @Myoldmopar I'd prefer a `switch` here too, but it won't work because `case` statements can't have a `const int` in them. To go that direction, I think we'd want `Bisection` etc. to be in an enum:

```
enum class HVACSystemRootFindingAlgorithm { RegulaFalsi, Bisection, 
    RegulaFalsiThenBisection, BisectionThenRegulaFalsi, Alternation };
```

and then the code can look like

```
switch( HVACSystemRootFinding.Type ) {
case HVACSystemRootFindingAlgorithm::Bisection:
  // do stuff
default:
  //do other stuff
}
```

We've talked about this in the past, and it is something we need to start doing. I don't know if now is the time, previously output issues held it up. My vote is "yes", since this is a new object. But I will leave it to you guys.